lib/plugins/aws/provider/awsProvider.js

```diff
@@ -470,7 +470,11 @@ class AwsProvider {
   }
 
   getDeploymentPrefix() {
-    return this.serverless.service.provider.deploymentPrefix || 'serverless';
+    const provider = this.serverless.service.provider;
+    if (!('deploymentPrefix' in provider)) {
```

Comment:
This checks for existence, so still `deploymentPrefix` set to values `null` or `undefined` will result with `'serverless'` prefix, which is not expected.

Technically the only way to make it right is: `customPrefix == null ? 'serverless' : customPrefix`

Reply:
Ah, got you; the plan was to allow those but not other falsey values. Whoops. :p

Reply:
Yes, we default to `'serverless'` only  for _non-values_ (`null` and `undefined`), falsy values would be in scope, if we will process property of _boolean_ type not _string_.